### ISSUE TITLE
Security threat. /bin/sh: 1: Syntax error due to improper character handling

### ISSUE THREAD

**Issue by emsi** [promptfoo/promptfoo#339]
When using exec provider with variable with special characters. eg:
```
prompts:
  - "{{violation}}"


providers: 
  - exec:python3 provider.py:

tests:
  - description: exploit
    vars:
      violation: "`('"
``` 

Whenever a variable contains special shell characters we might get an error like:
```
Error: Command failed: python3 provider.py "[{\"role\":\"user\",\"content\":\"`('\"}]" "{\"id\":\"exec:python3 provider.py\",\"config\":{\"basePath\":\"\"}}" "{\"vars\":{\"violation\":\"`('\"}}"
/bin/sh: 1: Syntax error: Unterminated quoted string
```

**Using shell when executing script providers is also a security threat! The provider should be executed directly (via exec call) rather that via system call  through shell.** Adding more escaping rules is not the right path as it can quickly get convoluted and either brake the passed values or leave a gap for another escaping trick.
For example if the variable is:
```
      violation: "`rm -rf /*`"
```

It will execute the `rm -rf /*` command rather than passing that to provider. This behavior is not only insecure but also inconsistent with other providers.

**Comment by emsi** [comment]
I believe in JS it's `spawn()` that does the proper `exec` syscall (and JS's `exec` is actually a `system` call):

https://stackabuse.com/executing-shell-commands-with-node-js/

**Comment by typpo** [comment]
Thanks for raising this.  #343 rewrites exec calls to not use a shell via `execFile` and `spawn`.  You're right that this solves a lot of headaches and potential side effects with escaping.

There is a backwards compatibility issue for anyone who relied on shell behaviors.  I'm considering retaining support with a `useShellUnsafe` or similar config param for the `exec` provider type.

**Comment by typpo** [comment]
Resolved in version 0.33.0
